### PR TITLE
[3456] Draft records last updated sort is inaccurate

### PR DESCRIPTION
**Pull request by AbigailMcP**
### Context

https://trello.com/c/TURjhLuH/3456-draft-records-last-updated-sort-is-inaccurate

### Changes proposed in this pull request

It _looked_ like the drafts weren't being ordered by last updated it.
They actually were. It was the 'Updated' timestamp on the trainee row that was wrong.
We decided to use the last entry on the timeline for this date so as not to confuse users with 'behind the scenes updates'.
But for draft trainees we don't _show_ any timeline events! So this date was always showing created_at, regardless of updates.

This PR:
- Uses the `updated_at` for drafts rather than the (empty) timeline
- Removes method `ordered_by_drafts_then_by` - relic from when drafts and registered were on same page
- Removes method `ordered_by_drafts` - as above
- Removes method `ordered_by_drafts_clause` - as above
- Removes unused method `ordered_trainees`
- Tidies up `application_record_card/view_spec.rb` to use factories and traits

### Guidance to review

- View the draft trainees
- Make a change to a trainee (not the first one)
- Save it and see that this is now first in the list AND the updated at is reflected on the card

### Important business

~* Does this PR introduce any PII fields that need to be overwritten or deleted in db/scripts/sanitise.sql?~
~* Does this PR change the database schema? If so, have you updated the config/analytics.yml file and considered whether you need to send 'import_entity' events?~

### CODE REVIEW

**Comment by AbigailMcP** [comment]
[draft records last updated sort is inaccurate](https://trello.com/c/TURjhLuH/3456-draft-records-last-updated-sort-is-inaccurate)

File: app/components/application_record_card/view.rb [hunk 1]
```diff
@@ -36,13 +36,18 @@ def route
...
+      date_text = tag.span(last_updated_date.strftime("%-d %B %Y"))
       class_list = "govuk-caption-m govuk-!-font-size-16 govuk-!-margin-top-2 govuk-!-margin-bottom-0 application-record-card__submitted"
 
       sanitize(tag.p(date_text.prepend("Updated: "), class: class_list))
     end
 
+    def last_updated_date
+      return record.updated_at if record.draft?
+
+      record.timeline&.first&.date.presence || record.updated_at
```
Comment (HettieS) [inline app/components/application_record_card/view.rb:48]: do we use `&.` here to handle nil values?

**Review by HettieS — APPROVED** [review]
LGTM! Had a play on the review app too and seems to be working great 👍

File: app/components/application_record_card/view.rb (hunk 1 above)
Comment (AbigailMcP) [reply app/components/application_record_card/view.rb:48]: yep! i haven't changed this part of the code from the old line 39 above - but it's guarding against a case where a trainee doesn't have a timeline.

i can't actually think of an 'everyday' reason why they wouldn't have one, but i would rather keep the code working as it has been for now!

**Pull request merged**